### ISSUE TITLE
Set-linger and other socket configs

### ISSUE THREAD

**Issue by ustulation** [carllerche/mio#566]
Could we have [set_buffer_sizes](https://doc.rust-lang.org/net2-rs/net2/trait.TcpStreamExt.html#tymethod.set_send_buffer_size), [set_linger](https://doc.rust-lang.org/net2-rs/net2/trait.TcpStreamExt.html#tymethod.set_linger) etc for live sockets like we have for `std::net` counterparts through `net2` crate ? Right now if i need to change anything on a running socket, i need to use unsafe functions to obtain raw-fd, wrap it to `std::net::TcpStream/UdpSocket`, perform the changes, get the `mio` equivalent again from raw-fd.

So basically the extensions provided by `net2` crate.

**Comment by carllerche** [comment]
cc @alexcrichton 

Should we impl `TcpStreamExt` on our `TcpStream`?

**Comment by alexcrichton** [comment]
Oh dear please no, those traits should never be implemented externally!

We should just implement the methods on the mio sockets :)

**Comment by carllerche** [comment]
Sounds good to me!
